form/persistence/persistence.cpp

```diff
@@ -2,78 +2,123 @@
 
 #include "persistence.hpp"
 
+#include <algorithm>
 #include <cstring>
 #include <utility>
 
 using namespace form::detail::experimental;
 
 // Factory function implementation
 namespace form::detail::experimental {
-  std::unique_ptr<IPersistence> createPersistence()
+  std::unique_ptr<IPersistence> createPersistence(
+    const form::experimental::config::parse_config& config) //factory takes form config
   {
-    return std::unique_ptr<IPersistence>(new Persistence());
+    return std::unique_ptr<IPersistence>(new Persistence(config));
```

Comment:
If you're adjusting this line of code anyway, I suggest using `std::make_unique:`

```suggestion
    return std::make_unique<Persistence>(config);
```